dashboard/app/controllers/pd/session_attendance_controller.rb

```diff
@@ -42,8 +42,20 @@ def select_enrollment
       return
     end
 
-    enrollment.update!(user: current_user, email: current_user.email)
+    enrollment.update!(user: current_user)
     Pd::Attendance.find_restore_or_create_by!(session: @session, teacher: current_user, enrollment: enrollment)
+
+    if current_user.student?
+      if Digest::MD5.hexdigest(enrollment.email) == current_user.hashed_email
```

Comment:
Nice. I didn't know that existed. I'll change it [here](https://github.com/code-dot-org/code-dot-org/blob/824fc7d76e96b256ad2b7204b93f538ed02b6f4a/dashboard/app/controllers/pd/workshop_enrollment_controller.rb#L181) too.